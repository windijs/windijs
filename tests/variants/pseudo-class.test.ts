import { $default, active, backgroundClip, backgroundClipConfig, backgroundColor, bundle, checked, colors, createUtility, empty, enabled, even, evenOfType, first, firstOfType, focus, focusVisible, focusWithin, hover, indeterminate, invalid, last, lastOfType, link, notChecked, notDisabled, notFirst, notFirstOfType, notLast, notLastOfType, notOnlyChild, notOnlyOfType, odd, oddOfType, onlyChild, onlyOfType, optional, placeholderShown, readOnly, readWrite, required, root, target, valid, visited } from "../../src";

const bg = createUtility("bg")
  .use(backgroundColor(colors))
  .use(backgroundClip(backgroundClipConfig))
  .init();

const utilities = [bg.blue[500], bg.clip.content];

test("first", () => {
  expect(bundle(first(...utilities))).toMatchSnapshot();
});

test("last", () => {
  expect(bundle(last(...utilities))).toMatchSnapshot();
});

test("odd", () => {
  expect(bundle(odd(...utilities))).toMatchSnapshot();
});

test("even", () => {
  expect(bundle(even(...utilities))).toMatchSnapshot();
});

test("visited", () => {
  expect(bundle(visited(...utilities))).toMatchSnapshot();
});

test("checked", () => {
  expect(bundle(checked(...utilities))).toMatchSnapshot();
});

test("focusWithin", () => {
  expect(bundle(focusWithin(...utilities))).toMatchSnapshot();
});

test("hover", () => {
  expect(bundle(hover(...utilities))).toMatchSnapshot();
});

test("focus", () => {
  expect(bundle(focus(...utilities))).toMatchSnapshot();
});

test("focusVisible", () => {
  expect(bundle(focusVisible(...utilities))).toMatchSnapshot();
});

test("active", () => {
  expect(bundle(active(...utilities))).toMatchSnapshot();
});

test("link", () => {
  expect(bundle(link(...utilities))).toMatchSnapshot();
});

test("target", () => {
  expect(bundle(target(...utilities))).toMatchSnapshot();
});

test("notChecked", () => {
  expect(bundle(notChecked(...utilities))).toMatchSnapshot();
});

test("enabled", () => {
  expect(bundle(enabled(...utilities))).toMatchSnapshot();
});

test("indeterminate", () => {
  expect(bundle(indeterminate(...utilities))).toMatchSnapshot();
});

test("invalid", () => {
  expect(bundle(invalid(...utilities))).toMatchSnapshot();
});

test("valid", () => {
  expect(bundle(valid(...utilities))).toMatchSnapshot();
});

test("optional", () => {
  expect(bundle(optional(...utilities))).toMatchSnapshot();
});

test("required", () => {
  expect(bundle(required(...utilities))).toMatchSnapshot();
});

test("placeholderShown", () => {
  expect(bundle(placeholderShown(...utilities))).toMatchSnapshot();
});

test("readOnly", () => {
  expect(bundle(readOnly(...utilities))).toMatchSnapshot();
});

test("readWrite", () => {
  expect(bundle(readWrite(...utilities))).toMatchSnapshot();
});

test("notDisabled", () => {
  expect(bundle(notDisabled(...utilities))).toMatchSnapshot();
});

test("firstOfType", () => {
  expect(bundle(firstOfType(...utilities))).toMatchSnapshot();
});

test("notFirstOfType", () => {
  expect(bundle(notFirstOfType(...utilities))).toMatchSnapshot();
});

test("lastOfType", () => {
  expect(bundle(lastOfType(...utilities))).toMatchSnapshot();
});

test("notLastOfType", () => {
  expect(bundle(notLastOfType(...utilities))).toMatchSnapshot();
});

test("notFirst", () => {
  expect(bundle(notFirst(...utilities))).toMatchSnapshot();
});

test("notLast", () => {
  expect(bundle(notLast(...utilities))).toMatchSnapshot();
});

test("onlyChild", () => {
  expect(bundle(onlyChild(...utilities))).toMatchSnapshot();
});

test("notOnlyChild", () => {
  expect(bundle(notOnlyChild(...utilities))).toMatchSnapshot();
});

test("onlyOfType", () => {
  expect(bundle(onlyOfType(...utilities))).toMatchSnapshot();
});

test("notOnlyOfType", () => {
  expect(bundle(notOnlyOfType(...utilities))).toMatchSnapshot();
});

test("root", () => {
  expect(bundle(root(...utilities))).toMatchSnapshot();
});

test("empty", () => {
  expect(bundle(empty(...utilities))).toMatchSnapshot();
});

test("evenOfType", () => {
  expect(bundle(evenOfType(...utilities))).toMatchSnapshot();
});

test("oddOfType", () => {
  expect(bundle(oddOfType(...utilities))).toMatchSnapshot();
});

test("default", () => {
  expect(bundle($default(...utilities))).toMatchSnapshot();
});
