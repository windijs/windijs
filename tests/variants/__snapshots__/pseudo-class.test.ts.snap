// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`active 1`] = `
Object {
  "&:active": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`checked 1`] = `
Object {
  "&:checked": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`default 1`] = `
Object {
  "&:default": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`empty 1`] = `
Object {
  "&:empty": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`enabled 1`] = `
Object {
  "&:enabled": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`even 1`] = `
Object {
  "&:nth-child(even)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`evenOfType 1`] = `
Object {
  "&:nth-of-type(even)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`first 1`] = `
Object {
  "&:first-child": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`firstOfType 1`] = `
Object {
  "&:first-of-type": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`focus 1`] = `
Object {
  "&:focus": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`focusVisible 1`] = `
Object {
  "&:focus-visible": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`focusWithin 1`] = `
Object {
  "&:focus-within": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`hover 1`] = `
Object {
  "&:hover": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`indeterminate 1`] = `
Object {
  "&:indeterminate": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`invalid 1`] = `
Object {
  "&:invalid": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`last 1`] = `
Object {
  "&:last-child": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`lastOfType 1`] = `
Object {
  "&:last-of-type": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`link 1`] = `
Object {
  "&:link": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`notChecked 1`] = `
Object {
  "&:not(:checked)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`notDisabled 1`] = `
Object {
  "&:not(:disabled)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`notFirst 1`] = `
Object {
  "&:not(:first-child)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`notFirstOfType 1`] = `
Object {
  "&:not(:first-of-type)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`notLast 1`] = `
Object {
  "&:not(:last-child)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`notLastOfType 1`] = `
Object {
  "&:not(:last-of-type)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`notOnlyChild 1`] = `
Object {
  "&:not(:only-child)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`notOnlyOfType 1`] = `
Object {
  "&:not(:only-of-type)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`odd 1`] = `
Object {
  "&:nth-child(odd)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`oddOfType 1`] = `
Object {
  "&:nth-of-type(odd)": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`onlyChild 1`] = `
Object {
  "&:only-child": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`onlyOfType 1`] = `
Object {
  "&:only-of-type": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`optional 1`] = `
Object {
  "&:optional": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`placeholderShown 1`] = `
Object {
  "&:placeholder-shown": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`readOnly 1`] = `
Object {
  "&:read-only": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`readWrite 1`] = `
Object {
  "&:read-write": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`required 1`] = `
Object {
  "&:required": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`root 1`] = `
Object {
  "&:root": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`target 1`] = `
Object {
  "&:target": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`valid 1`] = `
Object {
  "&:valid": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`visited 1`] = `
Object {
  "&:visited": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;
